fedimint-core/src/util/mod.rs

```diff
@@ -161,6 +169,20 @@ impl Display for SafeUrl {
     }
 }
 
+impl<'de> serde::de::Deserialize<'de> for SafeUrl {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s = SafeUrl(Url::deserialize(deserializer)?);
+
+        if s.port_or_known_default().is_none() {
+            return Err(serde::de::Error::custom("Invalid port"));
+        }
+
+        Ok(s)
```

Comment:
I'm not sure. Originally the derive macro would be calling `desieralize` on the `.0`, so that's what I did, and then I'm just fixing the port.

I bet `Url` parses a str, and then does it thing, so it's equivalent in a way, but this way is more mechanically corresponding to the original.

Reply:
just a bit of code duplication, that's why I asked